jetty-http/src/main/java/org/eclipse/jetty/http/UriCompliance.java

```diff
@@ -113,12 +112,11 @@ public String getDescription()
     @Deprecated
     public static final UriCompliance STRICT = new UriCompliance("STRICT", RFC3986.getAllowed());
 
-    private static final List<UriCompliance> KNOWN_MODES = Arrays.asList(DEFAULT, LEGACY, RFC3986, STRICT, SAFE);
     private static final AtomicInteger __custom = new AtomicInteger();
 
     public static UriCompliance valueOf(String name)
     {
-        for (UriCompliance compliance : KNOWN_MODES)
+        for (UriCompliance compliance : Arrays.asList(DEFAULT, LEGACY, RFC3986, STRICT, SAFE))
```

Comment:
Why not using `EnumSet.allOf()` or similar?

Reply:
Because they are not enums! They are just static final fields.
They were enums in 9, but that meant users could not define their own modes, so we have the CUSTOM[0-4] hack there.

Reply:
Doh! Ship it.